mod_reforged/hooks/msu.nut

```diff
@@ -8,6 +8,25 @@
 	return obj;
 }
 
+// Removes all duplicates from a string array and returns a new unique array
+::logInfo("Reforged::MSU -- adding ::MSU.Array.removeDuplicateStrings");
+::MSU.Array.removeDuplicateStrings <- function( _array )
+{
+	local _nonDuplicatesArray = [];
+	local _arrayTable = {};
+	foreach( string in _array )
+	{
+		::MSU.requireString(string);
+		_arrayTable[string] <- null;	// The value doesn't matter here. We just put in some random value
+	}
+
+	foreach( nonDuplicateString, value in _arrayTable )
```

Comment:
Use `::MSU.Table.keys` as that has a more optimized algorithm instead of this local `_nonDuplicatesArray`. You can simply `return ::MSU.Table.keys(_arrayTable);`